lib/ClangImporter/ImportDecl.cpp

```diff
@@ -7904,187 +7904,216 @@ createUnavailableDecl(Identifier name, DeclContext *dc, Type type,
-    auto table = findLookupTable(topLevelModule);
-    if (!table) return;
-
-    StringRef traceName;
-    if (topLevelModule)
-      traceName = topLevelModule->getTopLevelModuleName();
-    else
-      traceName = "(bridging header)";
-    PrettyStackTraceStringAction trace("loading import-as-members from",
-                                       traceName);
-    PrettyStackTraceDecl trace2("...for", nominal);
-
-    // Dig out the effective Clang context for this nominal type.
-    auto effectiveClangContext = getEffectiveClangContext(nominal);
-    if (!effectiveClangContext) return;
-
-    // Get ready to actually load the members.
-    ImportingEntityRAII Importing(*this);
-
-    // Load the members.
-    for (auto entry : table->lookupGlobalsAsMembers(effectiveClangContext)) {
-      auto decl = entry.get<clang::NamedDecl *>();
-
-      // Only include members in the same submodule as this extension.
-      if (getClangSubmoduleForDecl(decl) != submodule)
-        continue;
-
-      forEachDistinctName(decl, [&](ImportedName newName,
-                                    ImportNameVersion nameVersion) {
-        // Quickly check the context and bail out if it obviously doesn't
-        // belong here.
-        if (auto *importDC = newName.getEffectiveContext().getAsDeclContext())
-          if (importDC->isTranslationUnit())
-            return;
-
-        // Then try to import the decl under the specified name.
-        auto *member = importDecl(decl, nameVersion);
-        if (!member) return;
-
-        // Find the member that will land in an extension context.
-        while (!isa<ExtensionDecl>(member->getDeclContext())) {
-          auto nominal = dyn_cast<NominalTypeDecl>(member->getDeclContext());
-          if (!nominal) return;
-
-          member = nominal;
-          if (member->hasClangNode()) return;
-        }
-
-        if (member->getDeclContext() != ext) return;
-        ext->addMember(member);
-        
-        for (auto alternate : getAlternateDecls(member)) {
-          if (alternate->getDeclContext() == ext)
-            ext->addMember(alternate);
-        }
-      });
-    }
-
-    return;
+  if (auto objcContainer = dyn_cast_or_null<clang::ObjCContainerDecl>(D->getClangDecl()) ) {
+    loadAllMembersOfObjcContainer(D, objcContainer);
   }
+  else {
+    loadAllMembersIntoExtension(D, extra);
+  }
+}
 
-
+void ClangImporter::Implementation::loadAllMembersOfObjcContainer(Decl *D, const clang::ObjCContainerDecl *objcContainer) {
   clang::PrettyStackTraceDecl trace(objcContainer, clang::SourceLocation(),
                                     Instance->getSourceManager(),
                                     "loading members for");
-
   DeclContext *DC;
   IterableDeclContext *IDC;
-  SmallVector<ProtocolDecl *, 4> protos;
+  {
+    ExtensionDecl *ext;
+    figureOutTheDeclarationContextToImportInto(D, ext, DC, IDC);
+    if (ext)
+      loadMembersOfBaseImportedFromClang(ext);
+  }
```

Comment:
The more canonical way to do this would be to have ext be the result value and then do this:

if (auto *ext = figureOutTheDeclarationContextToImportInto(...)) {
   loadMembersOfBaseImportedFromClang(...)
}

Then you don't have the bare scope.